pkg/logging/logging.go

```diff
@@ -75,6 +77,27 @@ var (
 	logOptions = LogOptions{}
 )
 
+func init() {
+	log := DefaultLogger.WithField(logfields.LogSubsys, "klog")
+
+	// Make sure that klog logging variables are initialized so that we can
+	// update them from this file.
+	klog.InitFlags(nil)
+
+	// Make sure klog does not log to stderr as we want it to control the output
+	// of klog so we want klog to log the errors to each writer of each level.
+	flag.Set("logtostderr", "false")
+	// We don't need all headers because logurs will already print them if
+	// necessary.
+	flag.Set("skip_headers", "true")
+
+	klog.SetOutputBySeverity("INFO", log.WriterLevel(logrus.InfoLevel))
+	klog.SetOutputBySeverity("WARNING", log.WriterLevel(logrus.WarnLevel))
+	klog.SetOutputBySeverity("ERROR", log.WriterLevel(logrus.ErrorLevel))
+	// We don't want klog to fatal us so we will only error.
+	klog.SetOutputBySeverity("FATAL", log.WriterLevel(logrus.ErrorLevel))
```

Comment:
are we sure this is desired? what happens when a fatal is reached now?

Reply:
Yeah I was on the fence on this one. I've switched to also fatal if the severity is "FATAL"